src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -375,17 +375,22 @@ public static int[] GetProcessIds(string machineName, bool isRemoteMachine)
 
         public static int[] GetProcessIds()
         {
-            int[] processIds = new int[256];
+            int[] processIds = ArrayPool<int>.Shared.Rent(256);
```

Comment:
Why not stackalloc this one?

The general rule of thumb is that stack allocations `<= 1024 bytes` are "safe". This is what `Windows` uses for its own `_alloca` method (which dynamically picks `stackalloc` or `malloc`)

Reply:
> Why not stackalloc this one?

We could do the whole dance of starting with stackalloc'd memory and then growing with the pool if it's too small.

Reply:
No preference on my end. If we expect most cases to fit in 256, it might avoid an allocation (even if pooled)

Notably on my Win11 box, I have 373 processes running atm. So it's also possible that this "default" is too small for some common cases of modern machines and that negates the ability to stackalloc in the first place.

Reply:
I'm going to leave this as-is.  I believe we hit this only on a relatively rare fallback path.